tdrs-backend/tdpservice/settings/common.py

```diff
@@ -60,13 +60,12 @@ class Common(Configuration):
     # https://docs.djangoproject.com/en/2.0/topics/http/middleware/
     MIDDLEWARE = (
         "django.middleware.security.SecurityMiddleware",
-        "tdpservice.middleware.SessionMiddleware",
```

Comment:
This broke dev and staging.

Reply:
I am almost certain that by removing this and fixing the other middleware files I got further in the login flow.

However, it was not until I removed the proceeding lines  that the app started to work in both environments.

I thought that there were some changes in common.py that were added to fix production. When I was testing to see what would get staging to work I decided to to use the old version of common.py. I expected that to make staging  work, but break production. It ended up working on both environments. 

It could be that the changes in common.py are no longer needed  due to changes elsewhere in the code (the nginx config or env vars).

Reply:
this looks right to me at this point.. you've got  the cors middleware and django session middleware installed for all environments, then the custom production session middleware added to the production config.